### PR TITLE
fix a bug that cache entries were never updated

### PR DESCRIPTION
**Pull request by johnlanni**
**Please provide a description of this PR:**
 
The last touch time of the wasm cache entry was never updated.

### CODE REVIEW

File: pkg/wasm/cache.go [hunk 1]
```diff
@@ -226,6 +226,7 @@ func (c *LocalFileCache) getEntry(key cacheKey) string {
 	if ce, ok := c.modules[key]; ok {
 		// Update last touched time.
 		ce.last = time.Now()
+		c.modules[key] = ce
```
Comment (zirain) [inline pkg/wasm/cache.go:229]: why need this?

Comment (johnlanni) [reply pkg/wasm/cache.go:229]: the `ce` is a copy of the element in map

Comment (zirain) [reply pkg/wasm/cache.go:229]: ce is not a point?

Comment (johnlanni) [reply pkg/wasm/cache.go:229]: yes. you can see here:https://github.com/istio/istio/pull/37578/files#diff-b4b9abc466fa6e7cad54037be486711eda1582c289c1dcbdd13aebd776442f5cL94

Comment (zirain) [reply pkg/wasm/cache.go:229]: how about change it to a point, wirting back looks wired.

Comment (johnlanni) [reply pkg/wasm/cache.go]: ok

**Review by bianpengyuan — COMMENTED** [review]
Thanks! Can we add a unit test for this?

**Comment by johnlanni** [comment]
> Thanks! Can we add a unit test for this?

done

**Review by zirain — APPROVED** [review]
LGTM, thx!

**Pull request merged**